src/server/tran_server.cpp

```diff
@@ -443,25 +462,12 @@ tran_server::connection_handler::set_connection (cubcomm::channel &&chn)
   // Transaction server will use message specific error handlers.
   // Implementation will assert that an error handler is present if needed.
 
-  auto lockg_state = std::lock_guard<std::shared_mutex> { m_state_mtx };
+  assert (m_conn == nullptr);
+  m_conn.reset (new page_server_conn_t (std::move (chn), get_request_handlers (), tran_to_page_request::RESPOND,
+					page_to_tran_request::RESPOND, RESPONSE_PARTITIONING_SIZE, std::move (no_transaction_handler)));
+  assert (m_conn != nullptr);
```

Comment:
Kind of pointless assert.
```suggestion
```
I know it existed before, but for what?

Reply:
I don't know either. It even doesn't throw an exception. Let's remove it.